### PR TITLE
feat: add console-logging doc

### PR DESCRIPTION
**Pull request by alanmengg**
Signed-off-by: guiping meng <alan.meng@autocore.ai>

## Description

This pr add console-logging.md in autoware documentation.
Resolves https://github.com/autowarefoundation/autoware-documentation/issues/212.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/contributing/coding-guidelines/ros-nodes/console-logging.md [hunk 1]
```diff
@@ -1,5 +1,128 @@
 # Console logging
 
-!!! warning
+ROS2 logging is a powerful tool for understanding and debugging ros node.
```
Comment (shmpwk) [inline docs/contributing/coding-guidelines/ros-nodes/console-logging.md:3]:
```suggestion
ROS 2 logging is a powerful tool for understanding and debugging ros node.
```

File: docs/contributing/coding-guidelines/ros-nodes/console-logging.md [hunk 2]
```diff
@@ -1,5 +1,128 @@
...
 
-    Under Construction
+The logging subsystem in ROS 2 delivers logging messages to a variety of targets, including console, disk and ROS2 network.
```
Comment (shmpwk) [inline docs/contributing/coding-guidelines/ros-nodes/console-logging.md:5]:
```suggestion
The logging subsystem in ROS 2 delivers logging messages to a variety of targets, including console, disk and ROS 2 network.
```

**Review by shmpwk — APPROVED** [review]

File: docs/contributing/coding-guidelines/ros-nodes/console-logging.md [hunk 3]
```diff
@@ -1,5 +1,128 @@
...
+
+Set severity level to `DEBUG` only outputs message in debug mode.
+
+`condition` choices are:
+
+- `THROTTLE` - output the given printf-style message no more than the given rate
+- `ONCE` - output the given printf-style message only the first time this line is hit
+- `EXPRESSION` - output the given printf-style message only if the given expression is true
+- `FUNCTION` - output the given printf-style message only if the given function returns true
+- `SKIPFIRST` - output the given printf-style message all but the first time this line is hit
```
Comment (kenji-miyake) [inline docs/contributing/coding-guidelines/ros-nodes/console-logging.md:34]: I feel this part can be delegated to the ROS 2 Documentation.
@alanmengg What is the reason that you want to write it here?

```suggestion
```

File: docs/contributing/coding-guidelines/ros-nodes/console-logging.md [hunk 4]
```diff
@@ -1,5 +1,128 @@
...
+
+- {function_name} - The function name this was called from (may be empty).
+
+- {file_name} - The file name this was called from (may be empty).
+
+- {time} - The time in seconds since the epoch.
+
+- {time_as_nanoseconds} - The time in nanoseconds since the epoch.
+
+- {line_number} - The line number this was called from (may be empty)
```
Comment (kenji-miyake) [inline docs/contributing/coding-guidelines/ros-nodes/console-logging.md:60]: It's already written in https://autowarefoundation.github.io/autoware-documentation/main/installation/additional-settings-for-developers/.

Could you update it if you feel it's unclear?

```suggestion
```

**Review by kenji-miyake — CHANGES_REQUESTED** [review]
@alanmengg Thank you for your proposal.
I'm sorry, but I wanted to write documents such as "Don't output too many logs the console." or "Wait for more than 5 seconds when you initialize the node and are waiting the data to be ready.".

So, can I propose a document and could you wait for that, please? I'm sorry for my late work.
Since your example and tips are pretty good, I'd like to reflect them to my proposal. If you want, I can merge them first.

**Comment by alanmengg** [comment]
> @alanmengg Thank you for your proposal. I'm sorry, but I wanted to write documents such as "Don't output too many logs the console." or "Wait for more than 5 seconds when you initialize the node and are waiting the data to be ready.".
> 
> So, can I propose a document and could you wait for that, please? I'm sorry for my late work. Since your example and tips are pretty good, I'd like to reflect them to my proposal. If you want, I can merge them first.

 thank u for your appreciation. of course, I can wait for the  proposal.

> @alanmengg Thank you for your proposal. I'm sorry, but I wanted to write documents such as "Don't output too many logs the console." or "Wait for more than 5 seconds when you initialize the node and are waiting the data to be ready.".
> 
> So, can I propose a document and could you wait for that, please? I'm sorry for my late work. Since your example and tips are pretty good, I'd like to reflect them to my proposal. If you want, I can merge them first.

 thank u for your appreciation. of course, I can wait for the  proposal.

**Comment by alanmengg** [comment]
> @alanmengg Thank you for your proposal. I'm sorry, but I wanted to write documents such as "Don't output too many logs the console." or "Wait for more than 5 seconds when you initialize the node and are waiting the data to be ready.".
> 
> So, can I propose a document and could you wait for that, please? I'm sorry for my late work. Since your example and tips are pretty good, I'd like to reflect them to my proposal. If you want, I can merge them first.

 thank u for your appreciation. of course, I can wait for the  proposal.

**Comment by kenji-miyake** [comment]
TODO(@kenji-miyake): Consider https://github.com/orgs/autowarefoundation/discussions/2824#discussioncomment-3967388 and https://github.com/autowarefoundation/autoware.universe/pull/2056. cc @xmfcx

**Comment by kenji-miyake** [comment]
@alanmengg Hi, I'm so sorry to be late for my work. I had to do a lot of in-company work. :cry: 
I've submitted my proposal to this page #283. Could you review it?

Regarding your updates 406b3ea83a002ab17a0c75c7130465281254ff0b, I'll check it now.

**Comment by kenji-miyake** [comment]
@alanmengg Oh, your updates look simple and nice! :smile: 
But I believe they are almost covered by my proposal #283 and it has a little more content. So, would it be possible for you to add your proposal to my proposal?

**Comment by alanmengg** [comment]
> @alanmengg Oh, your updates look simple and nice! 😄 But I believe they are almost covered by my proposal #283 and it has a little more content. So, would it be possible for you to add your proposal to my proposal?

I‘ve read your  proposal, it's great, and it would be nice to combine them togther, but I don't know how to add my proposal to yours😢

**Comment by kenji-miyake** [comment]
@alanmengg Thank you for your feedback.
You can add review comments to https://github.com/autowarefoundation/autoware-documentation/pull/283, or update this PR after https://github.com/autowarefoundation/autoware-documentation/pull/283 is merged.
Any way you prefer is okay for me.

**Comment by alanmengg** [comment]
> @alanmengg Thank you for your feedback. You can add review comments to #283, or update this PR after #283 is merged. Any way you prefer is okay for me.
I directly add review comments in #283, but sincerely there's not much to add after i review it thoroughly. Thank u for the great content!👍 Should I close this pr?

**Comment by kenji-miyake** [comment]
@alanmengg Yes, if you feel #283 covers all content, you can close this.

**Comment by alanmengg** [comment]
As #283 covers mostly and more content than this pr. It is closed now

**Pull request closed without merging**